src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -247,12 +324,21 @@ private static void Report(Exception exception, Action<Exception>? handler)
                 return;
             }
 #endif
+
             if (!exception.Data.IsReadOnly)
             {
                 exception.Data[s_reportedMarker] = s_reportedMarker;
             }
 
-            handler.Invoke(exception);
+            s_handler(exception, severity, forceDump);
         }
     }
+
+    internal enum ErrorSeverity
+    {
+        Uncategorized,
```

Comment:
"Uncategorized" is the default for now, but the VS APIs recommends not using that for anything new. My plan is to merge this PR, do another cycle or two getting everything else annotated, and then removing "Uncategorized" entirely along with the defaults of the parameters. I don't want to bloat this PR _too_ much annotating everything, and also doing the switch to remove the defaults will break in-flight PRs trying to call the Report APIs so we need to do that quickly.